lib/grandpa/message_handler.go

```diff
@@ -215,10 +227,16 @@ func (h *MessageHandler) handleCatchUpResponse(msg *CatchUpResponse) error {
 
 	prevote, err := h.verifyPreVoteJustification(msg)
 	if err != nil {
+		// no need to check for chaindb.ErrKeyNotFound
+		// Since highest block was present (otherwise execution would not reach here),
+		// we expect previous blocks to exists as well. If they don't exist, we should error.
```

Comment:
@noot This is what i inferred from the spec but I ain't super sure about this. Need confirmation!

So `msg.Hash` is highest ancestor of all blocks voted on in the arrays of justifications. I am assuming that we sync sequentially(true?). So, if we have the highest block we must have previous blocks, and thus ok to error. Same for verifyPreCommitJustification.

Reply:
addressed